### PR TITLE
KAFKA-6474: Rewrite tests to use new public TopologyTestDriver [part 3]

### PR DESCRIPTION
**Pull request by h314to**
This PR is a further step towards the complete replacement of `KStreamTestDriver` with `TopologyTestDriver`. These straightforward changes were split from another [PR](https://github.com/apache/kafka/pull/4986) to simplify the review process.

* Refactor:
  - KStreamWindowReduceTest
  - KTableMapKeysTest
  - SessionWindowedKStreamImplTest
  - TimeWindowedKStreamImplTest

### CODE REVIEW

**Comment by mjsax** [comment]
\cc @bbejeck @vvcephei

**Review by mjsax — APPROVED** [review]
LGTM. Thanks for the PR!

Waiting for second review and Jenkins to pass before merging.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImplTest.java
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImplTest.java:156]: I think some of the above `@SuppressWarnings("unchecked")` can be removed now as well; could you double check?

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImplTest.java
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImplTest.java:155]: Ditto here for `@SuppressWarnings("unchecked")` above?

**Review by guozhangwang — APPROVED** [review]
Left a couple comments on checking `@SuppressWarnings("unchecked")`. Otherwise LGTM!

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImplTest.java
Comment (h314to) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImplTest.java:156]: Yes, they can. Thank for catching this.

**Comment by bbejeck** [comment]
retest this please

**Review by bbejeck — APPROVED** [review]
Thanks for the PR! LGTM.

**Comment by h314to** [comment]
I keep getting an (unrelated) timeout error on `
kafka.server.DynamicBrokerReconfigurationTest.testUncleanLeaderElectionEnable` with JDK10 and Scala 2.12

**Comment by guozhangwang** [comment]
@h314to This is a known issue and is being fixed.

**Comment by guozhangwang** [comment]
Merged to trunk, thanks @h314to !

**Comment by h314to** [comment]
Great! :)

**Pull request merged**
